### PR TITLE
Add IOU Menu Options

### PR DESCRIPTION
**Pull request by marcaaron**
### Details
cc @Julesssss @iwiznia adds the menu options behind beta

### Fixed Issues
Fixes https://github.com/Expensify/Expensify/issues/161984

### QA Steps
1. Login and tap the FAB menu +
2. Verify that the two new options appear
Screen Shot 2021-04-28 at 7 43 35 AM

3. Verify that clicking each one will bring you to the associated view
4. Select a chat with a single participant
5. Tap the `+` menu next to the comment compose area
6. Verify there is an option to `Request Money`
7. Tap and verify the request money screen appears and the participant is already selected
8. Select a chat with multiple participants
9. Tap the `+` menu next to the comment compose area
10. Verify there is an option to `Split Bill`
11. Tap and verify the split bill screen appears and all participants are already selected on the next screen
12. Log in with an account that is not under this beta and verify that the menus look as they did before and the new options are not present.

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

#### Web
Screen Shot 2021-04-28 at 7 43 35 AM

### CODE REVIEW

**Comment by marcaaron** [comment]
Will add screenshots and test on all platforms in a bit but wanted to get this into review sooner than later.

**Comment by francoisl** [comment]
Code looks good but when I create a split bill from a convo with multiple people and click the `Split Bill` option, the panel doesn't populate with anything other than the currency, and the Next button is disabled. Nothing in the JS console or logs, any idea?

**Comment by marcaaron** [comment]
>Code looks good but when I create a split bill from a convo with multiple people and click the Split Bill option, the panel doesn't populate with anything other than the currency, and the Next button is disabled. Nothing in the JS console or logs, any idea?

This PR shouldn't really have any effects on the way things are working there. But I think if you click to the right of the USD then you can enter a number and should be able to "Next" after that. I found this a little confusing myself as the view doesn't really look like an input of any kind so it's unclear what we can do on desktop.

**Comment by francoisl** [comment]
Oh derp, got it... I was expecting the view would directly show the bill participants.

**Review by francoisl — DISMISSED** [review]

File: src/pages/home/report/ReportActionCompose.js [hunk 1]
```diff
@@ -324,6 +333,27 @@ class ReportActionCompose extends React.Component {
...
+                                                            : {
+                                                                icon: MoneyCircle,
+                                                                text: 'Request Money',
+                                                                onSelected: () => {
+                                                                    Navigation.navigate(
+                                                                        ROUTES.getIouRequestRoute(this.props.reportID),
+                                                                    );
+                                                                },
+                                                            },
+                                                    ] : []),
```
Comment (Julesssss) [inline src/pages/home/report/ReportActionCompose.js:356]: Interesting way to add additional items without needing logic defined outside of the prop. Nice!

**Review by Julesssss — CHANGES_REQUESTED** [review]
The issue tests well. 

One request, would you mind removing the other 'Hiding menu options'[ comment here](https://github.com/Expensify/Expensify.cash/blob/72a113f89265e8873a4ece0cf6a46e3a96d43a07/src/pages/home/report/ReportActionCompose.js#L370-L380), as this PR makes it irrelevant. 

Also, could you add a test case for confirming that the IOU beta flag is working as expected in the description? (I was able to test this locally by removing the `all` beta flag, but for QA we should probably use accounts with/without the flag to verify)

**Comment by marcaaron** [comment]
Thanks for spotting that @Julesssss! I have made that requested change and added a test step to verify the new options do not appear for non beta users.

**Review by francoisl — APPROVED** [review]

**Comment by trjExpensify** [comment]
Are we good to go here without @Julesssss re-review so we can get testing, as I think he's out for the evening now?

**Review by Julesssss — APPROVED** [review]
Yeah, this change will not invalidate the tests I ran earlier. Merging

**Comment by Julesssss** [comment]
I can't find the merge button on the GitHub app, but we're good to go. Feel free to merge

**Pull request merged**